experimental/InvariantTheory/src/InvariantTheory.jl

```diff
@@ -0,0 +1,654 @@
+        if denominator((maxx*elem)//minn) != 1
+            error("den not 1")
+        end
+        push!(new_gens_, numerator((maxx*elem)//minn))
+        V= Vector{FieldElem}[]
+    end
+    return new_gens_
+end
+
+
+function mu_star(new_rep_mat::AbstractAlgebra.Generic.MatSpaceElem)
+    mixed_ring_xy = parent(new_rep_mat[1,1])
+    n = ncols(new_rep_mat)
+    vars = matrix(mixed_ring_xy,1,n,[gens(mixed_ring_xy)[i] for i in 1:n])
+    new_vars = vars*transpose(new_rep_mat)
+    D = Dict([])
+    for i in 1:n
+        push!(D, gens(mixed_ring_xy)[i]=>new_vars[i])
+    end
+    return D
+end
+
+function reynolds_v_slm(elem::MPolyDecRingElem, new_rep_mat::AbstractAlgebra.Generic.MatSpaceElem, new_det::MPolyDecRingElem, m::Int)
+    D = mu_star(new_rep_mat)
+    mixed_ring_xy = parent(elem)
+    sum_ = mixed_ring_xy()
+    #mu_star: 
+    mons = collect(monomials(elem))
+    coeffs = collect(coefficients(elem))
+    for i in 1:length(mons)
+        k = mixed_ring_xy(1)
+        factors = factorise(mons[i])
+        for j in 1:length(factors)
+            if factors[j][2] != 0
+                neww = getindex(D, factors[j][1])
+                k = k*(neww^(factors[j][2]))
+            end
+        end
+        sum_ += coeffs[i]*k
+    end
+    t = needed_degree(sum_, m)
+    if !divides(t, m)[1]
+        return parent(elem)()
+    else
+        p = divexact(t, m)
+    end
+    #num = omegap_(p, new_det, sum_)
+    #den = omegap_(p, new_det, (new_det)^p)
+    #if !(denominator(num//den)==1)
+    #    error("denominator of reynolds not rational")
+    #end
+    #return numerator(num//den)
+    return reynolds_slm(sum_, new_det, p)
+end
+
+function reynolds_slm(elem::MPolyElem, det_::MPolyElem, p::Int)
+    num = omegap_(p,det_, elem)
+    den = omegap_(p,det_,det_^p)
+    if !(denominator(num//den)==1)
+        error("denominator of reynolds not rational")
+    end
+    return numerator(num//den)
+end
+
+function needed_degree(elem::MPolyDecRingElem, m::Int)
+    R = parent(elem)
+    n = numerator((ngens(R) - m^2)//2)
+    extra_ring, _= PolynomialRing(base_ring(R), "z"=>1:m^2)
+    mapp = hom(R,extra_ring, vcat([1 for i in 1:2*n], gens(extra_ring)))
+    return total_degree(mapp(elem))
+end
+
+#works #Unused.
+function omegap(p::Int, det_::MPolyDecRingElem, f::MPolyDecRingElem)
+    parent(det_) == parent(f) || error("Omega process ring error")
+    action_ring = parent(det_)
+    detp = (det_)^p
+    monos = collect(monomials(detp))
+    coeffs = collect(coefficients(detp))
+    h = action_ring()
+    for i in 1:length(monos)
+        exp_vect = exponent_vector(monos[i], 1)
+        x = f
+        for i in 1:length(exp_vect)
+            for j in 1:exp_vect[i]
+                x = derivative(x, gens(action_ring)[i])
+                if x == 0
+                    break
+                end
+            end
+        end
+        h += coeffs[i]*x
+    end
+    return h
+end
+
+#alternate function for omegap. Works the same.
```

Comment:
Then why is it here? Because it (or the other one) is faster/easier/nicer/... ?!

And what's the difference? Should be stated in the comment.